File: freecodecamp/2026-01-08.js
function isSortedIncreasing(arr, n) {
  let i = 0;

  while(i < (n - 1)) {
    if (arr[i] > arr[i+1]) {
      return false;
    }
    i++
  }

  return true;
}

function isSortedDecreasing(arr, n) {
  let i = 0;

  while(i < (n - 1)) {
    if (arr[i] < arr[i+1]) {
      return false;
    }
    i++
  }

  return true;
}

function isSorted(arr) {
  const n = arr.length;

  return isSortedIncreasing(arr, n) 
    ? "Ascending"
    : isSortedDecreasing(arr, n)
      ? "Descending"
      : "Not sorted";
}

File: freecodecamp/2026-01-08.py
def is_sorted(arr):
    isAsc = True
    isDesc = True
    
    n = len(arr)
    
    if n <= 1:
        return "Ascending"
    
    for i in range(0, n-1):
        if arr[i] > arr[i+1]:
            isAsc = False
        
        if arr[i] < arr[i+1]:
            isDesc = False
    
    if isAsc:
        return "Ascending"
    elif isDesc:
        return "Descending"
    else:
        return "Not sorted"
